crates/rpc/rpc-eth-api/src/helpers/block.rs

```diff
@@ -64,20 +63,10 @@ pub trait EthBlocks: LoadBlock {
         async move {
             let Some(block) = self.block_with_senders(block_id).await? else { return Ok(None) };
             let block_hash = block.hash();
-            let mut total_difficulty = self
-                .provider()
-                .header_td_by_number(block.number())
-                .map_err(Self::Error::from_eth_err)?;
-            if total_difficulty.is_none() {
-                // if we failed to find td after we successfully loaded the block, try again using
-                // the hash this only matters if the chain is currently transitioning the merge block and there's a reorg: <https://github.com/paradigmxyz/reth/issues/10941>
-                total_difficulty =
-                    self.provider().header_td(&block.hash()).map_err(Self::Error::from_eth_err)?;
-            }
 
             let block = from_block(
                 (*block).clone().unseal(),
-                total_difficulty.unwrap_or_default(),
+                Default::default(),
```

Comment:
you're right, but I'll need to make some changes,  such as changing [from_block_with_tx_hashes](https://github.com/mvares/reth/blob/85334698442bdc449935ea9727d8359ada6aee21/crates/rpc/rpc-types-compat/src/block.rs#L32), so that it doesn't use `total_difficulty`

Reply:
we need to change `from_block_full` also

cc @mattsse

Reply:
yep, we can remove this everywhere imo

Reply:
https://github.com/paradigmxyz/reth/pull/13303/commits/355bdf87e15142407d68e7959cf19a8087e314eb